src/liballoc/boxed.rs

```diff
@@ -717,49 +790,53 @@ impl<I: FusedIterator + ?Sized> FusedIterator for Box<I> {}
 #[rustc_paren_sugar]
 #[unstable(feature = "fnbox",
            reason = "will be deprecated if and when `Box<FnOnce>` becomes usable", issue = "28796")]
-pub trait FnBox<A> {
+pub trait FnBox<Args, A: Alloc + Default = Global> {
     type Output;
 
-    fn call_box(self: Box<Self>, args: A) -> Self::Output;
+    fn call_box(self: Box<Self, A>, args: Args) -> Self::Output;
 }
 
 #[unstable(feature = "fnbox",
            reason = "will be deprecated if and when `Box<FnOnce>` becomes usable", issue = "28796")]
-impl<A, F> FnBox<A> for F
-    where F: FnOnce<A>
+impl<Args, F, A: Alloc + Default> FnBox<Args, A> for F
+    where F: FnOnce<Args>
 {
     type Output = F::Output;
 
-    fn call_box(self: Box<F>, args: A) -> F::Output {
+    fn call_box(self: Box<F, A>, args: Args) -> F::Output {
         self.call_once(args)
     }
 }
 
 #[unstable(feature = "fnbox",
            reason = "will be deprecated if and when `Box<FnOnce>` becomes usable", issue = "28796")]
-impl<A, R> FnOnce<A> for Box<dyn FnBox<A, Output = R> + '_> {
+impl<Args, R, A: Alloc + Default> FnOnce<Args> for Box<dyn FnBox<Args, A, Output = R> + '_, A> {
     type Output = R;
 
-    extern "rust-call" fn call_once(self, args: A) -> R {
+    extern "rust-call" fn call_once(self, args: Args) -> R {
         self.call_box(args)
     }
 }
 
 #[unstable(feature = "fnbox",
            reason = "will be deprecated if and when `Box<FnOnce>` becomes usable", issue = "28796")]
-impl<A, R> FnOnce<A> for Box<dyn FnBox<A, Output = R> + Send + '_> {
+impl<Args, R, A: Alloc + Default> FnOnce<Args>
+    for Box<dyn FnBox<Args, A, Output = R> + Send + '_, A>
+{
     type Output = R;
 
-    extern "rust-call" fn call_once(self, args: A) -> R {
+    extern "rust-call" fn call_once(self, args: Args) -> R {
         self.call_box(args)
     }
 }
 
 #[unstable(feature = "coerce_unsized", issue = "27732")]
-impl<T: ?Sized + Unsize<U>, U: ?Sized> CoerceUnsized<Box<U>> for Box<T> {}
+impl<T: ?Sized + Unsize<U>, U: ?Sized, A: Alloc + Default> CoerceUnsized<Box<U, A>>
+    for Box<T, A> {}
 
 #[unstable(feature = "dispatch_from_dyn", issue = "0")]
-impl<T: ?Sized + Unsize<U>, U: ?Sized> DispatchFromDyn<Box<U>> for Box<T> {}
+impl<T: ?Sized + Unsize<U>, U: ?Sized, A: Alloc + Default> DispatchFromDyn<Box<U, A>>
+    for Box<T, A> {}
```

Comment:
As far as I understand, `DispatchFromDyn` is the only trait, which requires `A: Default`. It would also be possible to implement it like this:

```
impl<T: ?Sized + Unsize<U>, U: ?Sized> DispatchFromDyn<Box<U, Global>> for Box<T, Global> {}
```

and remove the `Default` bound.